File: Assets/Scripts/Hud.js
#pragma strict

var win:boolean = false;
var loss:boolean = false;
var width:int = 100;
var height:int = 20;

function OnGUI()
{
	if(win)
	{
		GUI.Label(Rect((Screen.width/2)-(width/2), (Screen.height/2)-(height/2), width, height), "YOU WIN!!!");
	}else if(loss){
		GUI.Label(Rect((Screen.width/2)-(width/2), (Screen.height/2)-(height/2), width, height), "YOU LOOSE!");
	}else{
		//Draw Hud
	}
}

File: Assets/Scripts/Hud.cs
using UnityEngine;
using System.Collections;

public class Hud : MonoBehaviour 
{
		
	bool gameOver = false;	
	
	public int width = 100;
	public int height = 20;
	
	Score score;
	
	//gui stuff
	Rect rect;
	
	void Start()
	{
		score = (Score) GetComponent("Score"); 
		rect = new Rect(Screen.width - 200, 0, 200,50);
	}
	
	bool OnGUI()
	{
		if(gameOver)
		{
			Application.LoadLevel("GameOver");
		}
		else
		{
			//Draw Hud
			if (!score)
			{
				Debug.LogError("No Score Model");
				return false;
			}
			
			GUI.Label( rect, "Score: " + score.GetScore().ToString());
		}
		
		return true;
	}
	
	public bool IsGameOver()
	{
		return gameOver;
	}
	
	public void GameOver()
	{
		gameOver = true;
	}
	
	public void Reset()
	{
		gameOver = false;
	}
}
